homeassistant/components/opengarage/cover.py

```diff
@@ -153,12 +162,30 @@ async def async_update(self):
             self._state = state
 
         _LOGGER.debug("%s status: %s", self._name, self._state)
-        if status.get("rssi") is not None:
-            self._extra_state_attributes[ATTR_SIGNAL_STRENGTH] = status.get("rssi")
         if status.get("dist") is not None:
             self._extra_state_attributes[ATTR_DISTANCE_SENSOR] = status.get("dist")
+        if status.get("sn2") is not None:
+            self._extra_state_attributes[ATTR_SWITCH_SENSOR] = status.get("sn2")
         if self._state is not None:
             self._extra_state_attributes[ATTR_DOOR_STATE] = self._state
+        if status.get("vehicle") is not None:
+            self._extra_state_attributes[ATTR_VEHICLE_STATUS] = status.get("vehicle")
+        if status.get("rcnt") is not None:
+            self._extra_state_attributes[ATTR_READ_COUNT] = status.get("rcnt")
+        if status.get("fwv") is not None:
+            self._extra_state_attributes[ATTR_FIRMWARE_VERSION] = status.get("fwv")
+        if status.get("name") is not None:
+            self._extra_state_attributes[ATTR_NAME] = status.get("name")
+        if status.get("mac") is not None:
+            self._extra_state_attributes[ATTR_MAC_ADDRESS] = status.get("mac")
+        if status.get("cid") is not None:
+            self._extra_state_attributes[ATTR_WIFI_CHIP_ID] = status.get("cid")
+        if status.get("rssi") is not None:
+            self._extra_state_attributes[ATTR_SIGNAL_STRENGTH] = status.get("rssi")
+        if status.get("temp") is not None:
+            self._extra_state_attributes[ATTR_TEMPERATURE] = status.get("temp")
+        if status.get("humid") is not None:
+            self._extra_state_attributes[ATTR_HUMIDITY] = status.get("humid")
```

Comment:
These should not be attributes, but sensors. It probably applies to others as well, as they provide a different state, thus should be their own sensors instead.

Reply:
Thanks @frenck.  In this case, since the integration is currently specifically set up as a `cover` platform, would the whole integration need to be overhauled (e.g. to move its settings to a top-level configuration.yaml object) or can sensor entities be created during the setup process for the `cover`?  I can't see an example of the latter happening in other integrations within the current Home Assistant code base.

If the integration needs an overhaul, then I'm not really in a position to make that happen right now. For this PR, however, I was adding those sensors out of convenience since I was writing the PR anyway, but not necessity for my own use case.  My main use case was getting what would be hopefully be considered actual attributes (device name, MAC, firmware version) - so if you're amenable, I can trim the PR down to just those attrs.

Reply:
In general, state attribute should rarely be used, almost all of them should be separate entities.

A refactor is needed in that case indeed, introducing them now, will be a breaking change later. We should not do that.